scylla-cdc-base/src/main/java/com/scylladb/cdc/model/worker/TaskState.java

```diff
@@ -108,6 +128,12 @@ public TaskState trimTaskState(Timestamp minimumWindowStart, long windowSizeMs)
             return new TaskState(minimumWindowStart, minimumWindowStart.plus(windowSizeMs, ChronoUnit.MILLIS), Optional.empty());
         }
 
-        return this;
+        // Trim the start of the window with minimumWindowStart.
+        Timestamp newWindowStart = windowStart;
```

Comment:
Previously it only handled one case:

```
---- WINDOW ---     | TTL

->

                    ---- WINDOW ---
```

Now it also handles this case:
```
----      WINDOW           ---     
                    | TTL
->

                    - WINDOW -     
                    | TTL
```

Previously it didn't really matter to do the second type of trimming, because the windows were very small (default 30 seconds).